airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -52,15 +52,27 @@ definitions:
       - emitted_at
     properties:
       stream:
-        description: "the name of the stream for this record"
+        description: Name of the stream for this record
         type: string
       data:
-        description: "the record data"
+        description: The data
         type: object
         existingJavaType: com.fasterxml.jackson.databind.JsonNode
       emitted_at:
-        description: "when the data was emitted from the source. epoch in millisecond."
+        description: When the data was emitted from the source. (epoch in milliseconds)
         type: integer
+      metadata:
+        "$ref": "#/definitions/AirbyteRecordMessageMetadata"
+  AirbyteRecordMessageMetadata:
```

Comment:
why not make it tighter? I am not super thrilled about making it it an undocumented struct with only strings.

Typically the well-known fields could be explicitly described here.

Reply:
i think this is what is being discussed in this thread [here](https://github.com/airbytehq/airbyte/pull/2522#discussion_r597243829)